great_expectations/experimental/datasources/spark_s3_datasource.py

```diff
@@ -0,0 +1,133 @@
+from __future__ import annotations
+
+import logging
+import re
+from typing import TYPE_CHECKING, Any, Dict, Optional, Union
+
+import pydantic
+from typing_extensions import Literal
+
+from great_expectations.core.util import S3Url
+from great_expectations.experimental.datasources import _SparkFilePathDatasource
+from great_expectations.experimental.datasources.data_asset.data_connector import (
+    DataConnector,
+    S3DataConnector,
+)
+from great_expectations.experimental.datasources.interfaces import TestConnectionError
+from great_expectations.experimental.datasources.spark_file_path_datasource import (
+    CSVAsset,
+)
+
+if TYPE_CHECKING:
+    from botocore.client import BaseClient
+
+    from great_expectations.experimental.datasources.interfaces import (
+        BatchSorter,
+        BatchSortersDefinition,
+    )
+
+
+logger = logging.getLogger(__name__)
+
+
+try:
+    import boto3
+except ImportError:
+    logger.debug("Unable to load boto3; install optional boto3 dependency for support.")
+    boto3 = None
+
+
+class SparkS3Datasource(_SparkFilePathDatasource):
+    # instance attributes
+    type: Literal["spark_s3"] = "spark_s3"
+
+    # S3 specific attributes
+    bucket: str
+    boto3_options: Dict[str, Any] = {}
+
+    _s3_client: Optional[BaseClient] = pydantic.PrivateAttr()
+
+    def __init__(self, **data):
+        super().__init__(**data)
+
+        try:
+            self._s3_client = boto3.client("s3", **self.boto3_options)
+        except (TypeError, AttributeError):
+            self._s3_client = None
```

Comment:
We can use `BOTO3_IMPORTED` to raise `SparkS3Datasource` error, like we did [here](https://github.com/great-expectations/great_expectations/blob/b9eeeb14f47cf754dabd4bb2cbcc3aa75aeb66d4/great_expectations/experimental/datasources/sql_datasource.py#L844). We can put that in a similar method called `self.get_s3_client` which can differentiate between missing dependency errors and incorrect configuration errors.

Reply:
@NathanFarmer Done!  Thanks!